pkgs/os-specific/linux/autosuspend/default.nix

```
@@ -0,0 +1,60 @@
+{ lib,
+  fetchFromGitHub,
+  python3
+}:
+
+let
+  pythonPackages = python3.pkgs;
+in
+
+pythonPackages.buildPythonApplication rec {
+  pname = "autosuspend";
+  version = "4.0.0";
+
+  src = fetchFromGitHub {
+    owner = "languitar";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "03qca6avn7bwxcavif7q2nqfzivzp0py7qw3i4hsb28gjrq9nz36";
+  };
+
+  propagatedBuildInputs = with pythonPackages; [
+    portalocker
+    psutil
+    dbus-python
+  ];
+
+  checkInputs = with pythonPackages; [
+    pytestCheckHook
+    python-dbusmock
+    pytest-httpserver
+    pytest-cov
+    dateutil
+    freezegun
+    pytest-mock
+    requests
+    requests-file
+    icalendar
+    tzlocal
+    jsonpath-ng
+    mpd2
+    lxml
+    pytest-datadir
+  ];
+
+  # Disable tests that need root
+  disabledTests = [
+    "test_smoke"
+    "test_multiple_sessions"
+  ];
+
+  doCheck = true;
+
+  meta = with lib ; {
+    description = "Periodically suspends a system on inactivity and wakes it up again automatically in case it is needed";
```

```suggestion
    description = "A daemon to automatically suspend and wake up a system";
```

Smaller description, taken from the website.
Usually those descriptions are displayed by `nix-env --query`; keeping them short (below the 80-char limit) helps when displayed in terminals.